homeassistant/components/rainbird/__init__.py

```diff
@@ -84,36 +63,46 @@
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Rain Bird component."""
 
     hass.data[DATA_RAINBIRD] = []
-    success = False
-    for controller_config in config[DOMAIN]:
-        success = success or _setup_controller(hass, controller_config, config)
 
-    return success
+    tasks = []
+    for controller_config in config[DOMAIN]:
+        tasks.append(_setup_controller(hass, controller_config, config))
+    return all(await asyncio.gather(*tasks))
 
 
-def _setup_controller(hass, controller_config, config):
+async def _setup_controller(hass, controller_config, config):
     """Set up a controller."""
     server = controller_config[CONF_HOST]
     password = controller_config[CONF_PASSWORD]
-    controller = RainbirdController(server, password)
+    client = AsyncRainbirdClient(async_get_clientsession(hass), server, password)
+    controller = AsyncRainbirdController(client)
     position = len(hass.data[DATA_RAINBIRD])
     try:
-        controller.get_serial_number()
-    except Exception as exc:  # pylint: disable=broad-except
+        await controller.get_serial_number()
+    except RainbirdApiException as exc:
         _LOGGER.error("Unable to setup controller: %s", exc)
         return False
     hass.data[DATA_RAINBIRD].append(controller)
+
+    rain_coordinator = RainbirdUpdateCoordinator(hass, controller.get_rain_sensor_state)
+    delay_coordinator = RainbirdUpdateCoordinator(hass, controller.get_rain_delay)
+
     _LOGGER.debug("Rain Bird Controller %d set to: %s", position, server)
     for platform in PLATFORMS:
         discovery.load_platform(
```

Comment:
We need to use the async API here, `discovery.async_load_platform`.

Reply:
Ah will do.